### PR TITLE
[util] build_docs.py check if existing hugo is hugo-extended

### LINKED ISSUE #757: [util] build_docs.py should check if existing hugo is hugo-extended
We require hugo-extended to build the docs for OpenTitan. Unfortunately, this still uses the `hugo` binary name. The logic added to build_docs.py will download hugo if it's not available. But the experience could be improved in the case that a `hugo` binary exists, but it's not hugo-extended.

build_docs.py should be updated to check `hugo version` output for the "extended" string, and to still download a copy of hugo if "extended" isn't found.

### PR DESCRIPTION
**Pull request by silvestrst**
build_docs.py uses hugo-extended to build the docs for OpenTitan, which
is stored as the binary named "hugo". When this binary is present the script
does not check whether the version is extended.

This change adds a hugo version check, which makes sure that the binary is
extended, and downloads the correct version if it is not.

Fixes #757
Signed-off-by: Silvestrs Timofejevs <silvestrst@lowrisc.org>

### CODE REVIEW

File: util/build_docs.py [hunk 1]
```diff
@@ -161,6 +162,17 @@ def generate_selfdocs():
...
 
+def is_hugo_extended():
+    args = [
+        "hugo",
+        "version"
+    ]
+    process = subprocess.run(args, text=True, stdout=subprocess.PIPE,
+                             check=True, cwd=str(SRCTREE_TOP))
+
+    return bool(re.search("v\d+\.\d+\.\d+-.*/extended", process.stdout))
```
Comment (imphil) [inline util/build_docs.py]: This should work, but I would go less fancy here, given that the version string doesn't seem to be particularly stable.
`return process.stdout.find("extended")`

If you keep the regex, please add a comment with the of "hugo version" to make the regex easier to understand.

File: util/build_docs.py [hunk 2]
```diff
@@ -240,6 +252,11 @@ def main():
     os.environ["PATH"] += os.pathsep + str(hugo_localinstall_dir)
 
     try:
+        if not is_hugo_extended():
+            logging.info("Hugo version is not extended. "
```
Comment (imphil) [inline util/build_docs.py]: wording nit: "Hugo extended (with SASS support) is required, but only non-extended version found. Installing local copy and re-trying."

File: util/build_docs.py (hunk 1 above)
Comment (silvestrst) [reply util/build_docs.py]: My thinking was that its potentially more likely that they could add some other "extended", which then could result in an incorrect behaviour. This way seems more robust as long as the version portion of the string stays the same. Probably fine either way really.

**Comment by imphil** [comment]
Thanks! Congrats on your first merged pull request! :rocket:

File: util/build_docs.py [hunk 3]
```diff
@@ -240,6 +254,12 @@ def main():
     os.environ["PATH"] += os.pathsep + str(hugo_localinstall_dir)
 
     try:
+        if not is_hugo_extended():
+            logging.info(
+                "Hugo extended (with SASS support) is required, but only non-extended version found."
```
Comment (gkelly) [inline util/build_docs.py]: Add a space after the period.  This will print as:

```
[...]version found.Installing local [...]
```

Comment (imphil) [reply util/build_docs.py]: @silvestrst Sorry I missed this. Since this PR is already merged, can you do a follow-up PR?

**Pull request merged**